engine/src/main/java/org/terasology/particles/ParticleData.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2016 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.particles;
+
+import org.terasology.math.geom.Vector2f;
+import org.terasology.math.geom.Vector3f;
+import org.terasology.math.geom.Vector4f;
+
+/**
+ * Data object to store the data of a single particle.
+ * Used internally for swapping operations.
+ */
+public final class ParticleData {
+    // scalars
+    public float energy;
+
+    // 2d vectors
+    public final Vector2f textureOffset = new Vector2f();
+
+    // 3d vectors
+    public final Vector3f position = new Vector3f();
+    public final Vector3f previousPosition = new Vector3f();
+    public final Vector3f velocity = new Vector3f();
+    public final Vector3f scale = new Vector3f();
+
+    // 4d vectors
+    public final Vector4f color = new Vector4f();
```

Comment:
You mean in the comment or in a custom data structure?

In general I think RGBA is more or less the default when talking about colors, but to specify it in the comment wouldn't hurt! 

I'd refrain from using a custom data structure though: particles should stay very light and it could be argued, perhaps at a later stage, they should be even lighter than Vector4f, i.e. float[4]. But at this stage I'd recommend to stick to things that are clear rather than prematurely optimized.